src/backend/pipeline/pipeline.py

```diff
@@ -67,7 +77,17 @@ def run(self, imgs: Mat | list[Mat]) -> Data:
         data.set(imgs)
 
         # Run the chain
-        self.head.run(data)
+        iterator = self.head
+        while iterator is not None:
+            # Run the module
+            data = iterator.run(data)
+            # Upload data to the cloud async
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            loop.run_until_complete(
+                iterator.upload(data, self.collection, self.bucket, self.base_url))
```

Comment:
I think `asyncio` is meant for [loop-based asynchronicity](https://stackoverflow.com/questions/29902908/what-is-the-difference-between-concurrent-futures-and-asyncio-futures) (e.g. web server). How about using `concurrent` in a similar fashion to what's in `ParallelModule`?